sklearn/utils/estimator_checks.py

```diff
@@ -1339,6 +1343,35 @@ def check_estimators_dtypes(name, estimator_orig):
                 getattr(estimator, method)(X_train)
 
 
+def check_estimators_preserve_dtypes(name, estimator_orig):
+
+    if not _safe_tags(estimator_orig, 'preserves_32bit_dtype'):
```

Comment:
Usually we don't implement in this way. We avoid to call the function if the tag is false. see above.

Reply:
Here it was done inside, so that we would have a list of estimators that fail and that needs fixing, as a temporary approach.

Reply:
I moved it now, as one needs to change the default value of the tag anyways to true to check if an estimator actually fails the test and is not just is skipped.
And after the meta issue is resolved one doesn't need to change anything.